### ISSUE TITLE
Validation: Present better error messages for required fields 

### ISSUE THREAD

**Issue by mrosvik** [Altinn/app-frontend-react#130]
## Description
_As a user I want a clear description of the requirements that are needed to complete the form._

We need to explain in the text what field the error is about. Example: "You need to fill out your email address" instead of "This field is required".

## Design
https://www.figma.com/file/Pvti3aRcOwH0k5Z7DrLznk/Arbeidsomr%C3%A5de%3A-Altinn-Studio-APPS?node-id=2545%3A50628

<img width="1055" alt="image" src="https://user-images.githubusercontent.com/1464915/158777458-c9788335-508b-41db-9f46-639f4a22a194.png">

## Possible solutions
- Use a convention where texts with the same id as the corresponding data model element is used instead of the generic error texts.
- Get the field-name from input label + generate requirements from json-schema?
UPDATE:
Displaying WHICH field should be completely doable, in that we can access information about the component and fetch the label text and add this in to the validation message. 
Displaying requirements for the field is not trivial to do consistently and well. Suggest that any such requirements that are important for the end user should be displayed elsewhere, f.ex. in a help text or description to the field and not in the error message.

We also need to consider how we handle required field validation, as we have some custom behaviours that are not present for other kinds of validation (showing a banner with information about required fields, which turns red when there are missing required fields). Currently we loop through the available validation messages for the current view and pick up any that are for missing required fields. This might be difficult to do once we implement this issue, so we need to decide on how we structure information about missing required fields. Some options:
- Continue as before, but make sure that the checks for if any required fields are missing in the view also handle more complex/detailed validation messages.
- Add some property to the fields validations, like `missing`, to be sidelined with `errors` and `warnings` that we can use as a flag, instead of looping through validation errors.
- Add a separate structure under `validations` in store to contain information about missing required fields for each view. Something similar to the `invalidData`-structure that we already have.

## Development tasks
- [x] Consider possible solutions above. As a minimum we should tell the user which field is required. For example "Du må fylle ut fødselsnummer. ". But could we also include the requirements? For example: "Du må fylle ut fødselsnummer (11 siffer uten mellomrom)". 
- [ ] Decide how to structure info about missing required fields.
- [ ] Implement solution - telling user which field is required in the error message by using field label text.

## Guidelines for app developers
For custom validation messages, we should remind the app developer that they need to include keywords relating to the label in the validation message. "Dette feltet er påkrevd" Is not allowed.
- [ ] Update [guidelines](https://docs.altinn.studio/app/design/guidelines/components/error-message/) for custom validation messages.

## Definition of done

- [ ] Documentation (docs.altinn.studio) is updated (if relevant)
  - [ ] Technical documentation
  - [ ] User documentation
- [ ] QA
- [ ] Manual test is complete (if relevant)
- [ ] Automated test is implemented (if relevant)
- [ ] All tasks in this userstory are closed (i.e. remaining tasks are moved to other user stories or marked obsolete)

**Comment by xmrsa** [comment]
Would it be possible to adjust the text "Du må fylle ut" based on the **type** of the element in the json-file for the page layout? F.ex. if the element is of type "RadioButtons" you could use something like "Velg et alternativ: ", and for type "Checkboxes" you could use "Velg minst ett alternativ: "

**Comment by Magnusrm** [comment]
@xmrsa The default error message is possible to override by setting the `"errorMessage"` property on a component in the layout. https://docs.altinn.studio/app/development/logic/validation/#custom-error-messages
